pipeline_dp/dp_computations.py

```diff
@@ -247,3 +254,56 @@ def compute_dp_mean(count: int, sum: float, dp_params: MeanVarParams):
                                  dp_params.noise_kind)
     dp_mean = dp_normalized_sum / dp_count + middle
     return dp_count, dp_mean * dp_count, dp_mean
+
+
+def compute_dp_var(count: int, sum: float, sum_squares: float,
+                   dp_params: MeanVarParams):
+    """Computes DP variance.
+
+    Args:
+        count: Non-DP count.
+        sum: Non-DP sum.
+        sum_squares: Non-DP sum of squares.
+        dp_params: The parameters used at computing the noise.
+
+    Raises:
+        ValueError: The noise kind is invalid.
+
+    Returns:
+        The tuple of anonymized count, sum, sum_squares and variance.
+    """
+    sum_middle = dp_params.middle()
+    sum_squares_middle = dp_params.middle_squares()
+
+    normalized_sum = sum - count * sum_middle
+    normalized_sum_squares = sum_squares - count * sum_squares_middle
+
+    l0_sensitivity = dp_params.l0_sensitivity()
+
+    # Splits the budget equally between the two mechanisms.
+    (count_eps, count_delta), (sum_eps, sum_delta), (
+        sum_squares_eps, sum_squares_delta) = equally_split_budget(
+        dp_params.eps, dp_params.delta, 3)
+
+    dp_count = _add_random_noise(count, count_eps, count_delta, l0_sensitivity,
+                                 dp_params.linf_sensitivity(
+                                     pipeline_dp.Metrics.COUNT),
+                                 dp_params.noise_kind)
+    dp_normalized_sum = _add_random_noise(normalized_sum, sum_eps, sum_delta,
+                                          l0_sensitivity,
+                                          dp_params.linf_sensitivity(
+                                              pipeline_dp.Metrics.MEAN),
+                                          dp_params.noise_kind)
+    dp_normalized_sum_squares = _add_random_noise(normalized_sum_squares,
+                                                  sum_squares_eps,
+                                                  sum_squares_delta,
+                                                  l0_sensitivity,
+                                                  dp_params.linf_sensitivity(
+                                                      pipeline_dp.Metrics.VAR),
+                                                  dp_params.noise_kind)
+
+    dp_mean = dp_normalized_sum / dp_count + sum_middle
+    dp_mean_squares = dp_normalized_sum_squares / dp_count + sum_squares_middle
+    dp_var = dp_mean_squares - dp_mean ** 2
```

Comment:
There's a lot of code duplication between this and compute_dp_mean. Can we share code? In particular, it seems like we should be able to have a function that takes:
- A raw sum
- A noisy count
- An upper and a lower bound

and use it to calculate the means in both variance and mean and the mean of squares in variance.